src/psij/executors/batch/template_function_library.py

```diff
@@ -0,0 +1,60 @@
+from datetime import timedelta
+from typing import Dict, Callable, Any, Union
+
+
+def walltime_to_minutes(walltime: Union[timedelta, int, str]) -> int:
+    """Converts a walltime object to a number of minutes.
+
+    The walltime can either be a Python :class:`~datetime.timedelta`, an integer, in which case
+    it is interpreted directly as a number of minutes, or a string with a format of either
+    `HH:MM:SS`, `HH:MM`, or `MM`.
+    Parameters
+    ----------
+    walltime
+        the walltime to convert
+    """
+    if isinstance(walltime, timedelta):
+        return int(walltime.total_seconds() / 60)
+    if isinstance(walltime, int):
+        return walltime
+    if isinstance(walltime, str):
+        return _parse_str_walltime(walltime)
+    raise ValueError('Invalid walltime: %s of type %s' % (walltime, type(walltime)))
+
+
+def _parse_str_walltime(walltime: str) -> int:
+    # accept HH:MM:SS or HH:MM or MM
+    els = walltime.split(':')
+    if len(els) > 3 or len(els) == 0:
+        raise ValueError('Invalid walltime format (%s). Expected HH:MM:SS or HH:MM or MM.' %
+                         walltime)
+    if len(els) == 3:
+        try:
+            return int(els[0]) * 60 + int(els[1])
```

Comment:
Minor nit: if `els[2] > 0` do we want to add 1 (i.e., take the ceiling)?

Reply:
yeah, maybe, but, like, meh